### DISCUSSION TITLE
PhoSpor - a MS Access based Phoniebox Spotify card management DB

### DISCUSSION THREAD

**Discussion by Micharius** [MiczFlor/RPi-Jukebox-RFID#2471] [Show and tell]
> Daddy can you make me a  card for my Phoniebox real quick?

In order to better manage the growing collection of cards for my daughter's ‘Phoniebox’, and above all to be able to create cards more quickly, I have made a solution. The whole thing is based on an Access database and is called ‘PhoSpor’ (short for ‘Phoniebox Spotify organiser’). Yes, quite oldschool with VBA etc., but it allowed me to create a useful solution in a relatively short time. 

PhoSpor currently has the following features:

- Drag and drop Spotify URLs from the browser
- Retrieval and filling of metadata such as title, artist, link to cover photo via connection to the Spotify API

Short video: 
https://github.com/user-attachments/assets/a2763b29-c120-4527-a062-bd3d006e2d57

- Automatic creation of the necessary folder in the ‘audiofolders’ of the ‘Phoniebox’ and the spotify.txt file contained therein
- Analysing the folders on the Phoniebox (what is in the DB, what is not)
- Import of existing ‘audiofolders’ folders and enrichment of the data with Spotify AP.
- Printing of labels (I currently use ‘Avery Zweckform 3481 with the Spotify metadata incl. image from Spotify)
- 
If anyone is interested, I am happy to share the DB with you. If there is more interest, I could also make a short manual. All you need is an ‘app’ set up on https://developer.spotify.com/ for yourself, client ID and secret can then be entered in PhoSpor so that the API works

Some pictures:

![pic1](https://github.com/user-attachments/assets/96df7b7a-7c80-4ae6-92fe-f0775ede6380)
![pic2](https://github.com/user-attachments/assets/42403d66-8035-49d5-bf52-439fd932ef37)
![pic3](https://github.com/user-attachments/assets/2f877ec7-3d1f-4060-94b3-ce0dbcd110b4)
![pic4](https://github.com/user-attachments/assets/c98ee317-bf69-48c4-9052-b41f6b1713b8)

**Comment by pabera** [comment]
Very, Very cool!
